api/models/User.js

```diff
@@ -161,6 +161,72 @@ module.exports = bookshelf.Model.extend(merge({
     .digest('hex')
   },
 
+  reactivate: function () {
+    return this.save({ active: true })
+  },
+
+  deactivate: async function (sessionId) {
+    Queue.classMethod('User', 'clearSessionsFor', { userId: this.get('user_id'), sessionId })
+    return this.save({ active: false })
+  },
+
+  sanelyDeleteUser: async function ({ sessionId, transacting = {} }) {
+    const deletedUserResult = await bookshelf.knex.raw('select id from users where email = "deleted@hylo.com"')
+    const deletedUserId = deletedUserResult.rows[0].id
+    Queue.classMethod('User', 'clearSessionsFor', { userId: this.get('user_id'), sessionId })
+    /* 
+      ### List of things to be done on account deletion ###
+
+      - zero out content of their posts and comments
+      - change the ownership of their posts and comments to the deletedUser record
+      - remove other references to their user_id
+      - delete their user record!
+    */
+
+    const query = `
+    BEGIN;
+    UPDATE posts SET name = 'Post by deleted user', description = '', active = false, location = NULL, location_id = NULL WHERE user_id = ${this.id};
```

Comment:
will active = false make it invisible to the front-end? is that what we want? we may want people to still see the deleted post with the existing comment thread on it

Reply:
Right, looks like it will hide them. I'll change.